spec/parser_spec.rb

```diff
@@ -1,45 +1,54 @@
-require_relative "../parser.rb"
-require 'json'
+# frozen_string_literal: true
 
+require_relative '../parser'
+require 'json'
 
 describe ParserGamesLogs do
-  let(:filename_with_path) { "games_logs_files/games.log" }
-  context "When the file exists" do
-    it "Instantiates the class." do
+  let(:filename_with_path) { 'spec/fixtures/test_file_games.log' }
+
+  context 'When the file exists' do
+    it 'Instantiates the class.' do
       expect { ParserGamesLogs.new(filename_with_path) }.not_to raise_error
     end
 
-
-    let(:first_line) { "  0:00 ------------------------------------------------------------\n" }
     describe '#read_first_line' do
+      let(:first_line) { "  0:00 ------------------------------------------------------------\n" }
+
       it "Returns the first line of the file 'games.log'" do
         parser = ParserGamesLogs.new(filename_with_path)
         parser_first_line = parser.read_first_line
         expect(parser_first_line).to eq(first_line)
       end
     end
 
-
-    let(:total_number_of_lines) {5306}
-    let(:players) {[
-                     "Isgalamido", "<world>", "Dono da Bola", "Mocinha", "Zeh", "Assasinu Credi",
-                     "Fasano Again", "Oootsimo", "UnnamedPlayer", "Maluquinho", "Mal",  "Chessus!",
-                     "Chessus"
-                   ]}
     describe '#parse_file' do
-      it "Returns a json object after parses the file." do
+      let(:total_number_of_lines) { 33 }
+      let(:players) do
+        [
+          'Isgalamido', 'Dono da Bola', 'Oootsimo', 'Assasinu Credi', 'Zeh', 'Mal'
+        ]
+      end
+      let(:kills) do
+        {
+          'Isgalamido' => 1, 'Dono da Bola' => 5, 'Oootsimo' => 3, 'Assasinu Credi' => 1, 'Zeh' => 2, 'Mal' => 0
+        }
+      end
+      let(:total_kills) { 16 }
+
+      it 'Returns a json object after parses the file.' do
         parser = ParserGamesLogs.new(filename_with_path)
         json_parsed_file = JSON.parse(parser.parse_file)
-        expect(json_parsed_file).to include("#{filename_with_path}" => { "lines"=> total_number_of_lines, "players"=> players })
+        expect(json_parsed_file).to include(filename_with_path.to_s => { 'lines' => total_number_of_lines,
+                                                                         'players' => players, 'kills' => kills, 'total_kills' => total_kills })
```

Comment:
Os lets são excelentes para construir um contexto e memoizá-los, mas não há ganhos em colocá-los do lado direito de uma expectation. a variável `parser` e a `json_parsed_file` são ótimas candidatas a virarem `lets`, pois elas descrevem um contexto e cai bem memoizá-las.

Em outras palavras, os lets preparam o contexto do seu teste, eles não servem pra ser a expectativa do seu teste (até podem ser usados com esse propósito, mas não como o único propósito, um exemplo de uso correto é o `filename_with_path.to_s`)

Mais informações em:

https://www.betterspecs.org/#let


```suggestion
        expect(json_parsed_file).to include(filename_with_path.to_s => {
          'lines' => 33,
          'players' => ['Isgalamido', 'Dono da Bola', 'Oootsimo', 'Assasinu Credi', 'Zeh', 'Mal'],
          'kills' => {'Isgalamido' => 1, 'Dono da Bola' => 5, 'Oootsimo' => 3, 'Assasinu Credi' => 1, 'Zeh' => 2, 'Mal' => 0},
          'total_kills' => 16 })
```